### ISSUE TITLE
Cannot be built from downloads tarball

### ISSUE THREAD

**Issue by hasufell** [luvit/luv#419]
lua and luajit submodules are missing from the tarball: https://github.com/luvit/luv/tree/master/deps

**Comment by squeek502** [comment]
Building from source should work with a [release tarball](https://github.com/luvit/luv/releases). Not much we can do about the Github zip not including submodules.

Are you using `LUA_BUILD_TYPE=System` when trying to build with system LuaJIT as described [here](https://github.com/luvit/luv#linking-with-shared-luajit)?

**Comment by hasufell** [comment]
> Building from source should work with a [release tarball](https://github.com/luvit/luv/releases)

Yes, I'm talking about the release tarball. I doesn't contain all submodules.

> Are you using `LUA_BUILD_TYPE=System` when trying to build with system LuaJIT as described [here](https://github.com/luvit/luv#linking-with-shared-luajit)?

Haven't tried that yet.

**Comment by squeek502** [comment]
> Yes, I'm talking about the release tarball. I doesn't contain all submodules.

Whoops, you're right. That tarball really only exists for LuaRocks, which doesn't expect to need to compile Lua from source. For compiling everything from source, your best best would be to do a git clone. Is there a reason you want a tarball instead?

**Comment by hasufell** [comment]
> Is there a reason you want a tarball instead?

Packaging. Using a git clone is suboptimal. The tarball should always have all submodules included.

**Comment by squeek502** [comment]
It's a bit harder of a call since this is a Lua module not tied to any specific Lua version. For packaging purposes, it should probably compile against whatever Lua version the user wants rather than compile LuaJIT from source, right?

Could you describe your use case a bit more? Are there other similar Lua projects that handle this sort of thing better?

**Comment by hasufell** [comment]
> For packaging purposes, it should probably compile against whatever Lua version the user wants rather than compile LuaJIT from source, right?

I'm on a source distro. Either I compile against the system version (there is only one, which is Luajit-2.0.5) or I use the bundled one. I cannot use the bundled one, because it's not in the tarball. The user cannot choose any version.

> Are you using `LUA_BUILD_TYPE=System` when trying to build with system LuaJIT as described [here](https://github.com/luvit/luv#linking-with-shared-luajit)?

This works

**Comment by squeek502** [comment]
I think this might just be down to default `make` being more development oriented than end-user oriented (or legacy-oriented, since `luv` started out as specific to LuaJIT and not a general Lua module). I don't think we should provide a bundled Lua with release distributions because, for example, if we bundled Lua 5.3 (which we also have as a submodule) and that was compiled/used by default, then the result would fail if you tried to run it with your system's LuaJIT.

Maybe we should move to `LUA_BUILD_TYPE=System` as more of a default.
